### PR TITLE
workflow: integrate fuzzy matcher in the `article` workflow

### PR DESCRIPTION
**Pull request by ammirate**
## Description

Changed the step in the workflow that checks if the ingested article is an update, by
performing a fuzzy match. If there is at least a match, the workflow halts, waiting
for human matching confirmation.

All the matches are now stored in `obj.extra_data.matches`, which is a dictionary containing
`exact` matches, `fuzzy` matches, and eventually the `approved` fuzzy match.

A new action called `MatchApproval` has been added to resume the workflow when is halted for
matching approval.

Solves #2527.

Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/matching.py [hunk 1]
```diff
@@ -105,12 +105,83 @@ def article_exists(obj, eng):
...
+                                'path':  'abstracts',
+                            },
+                            {
+                                'boost': 10,
+                                'path':  'authors[:3]',
+                            },
+                            {
+                                'boost': 10,
+                                'path':  'titles',
+                            },
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: @jmartinm derived a different configuration: https://github.com/inspirehep/invenio-matcher-benchmark/blob/83caef0aaddb59269cb4769fd1fc9d72fb8234f2/config.py

File: inspirehep/modules/workflows/tasks/matching.py [hunk 2]
```diff
@@ -98,12 +98,87 @@ def article_exists(obj, eng):
...
+            {
+                'queries': [
+                    {
+                        'clauses': [
+                            {
+                                'boost': 20,
+                                'path':  'abstracts',
+                            },
+                            {
+                                'boost': 10,
```
Comment (ammirate) [inline inspirehep/modules/workflows/tasks/matching.py]: @jacquerie I have updated the configuration tin according to https://github.com/inspirehep/invenio-matcher-benchmark/blob/83caef0aaddb59269cb4769fd1fc9d72fb8234f2/config.py, but removing the `boost` from `authors` I couldn't match the basic record I am using. Shall we go for some benchmarks? If yes, I think they are out of the scope of this PR.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 3]
```diff
@@ -98,12 +98,87 @@ def article_exists(obj, eng):
...
+    return bool(record_ids)
+
+
+@with_debug_logging
+def is_fuzzy_match_approved(obj, eng):
+    """Check if a fuzzy match has been approved.
+
+    Also, if the property `fuzzy_match_approved_id` in obj.extra_data
+    contains an id, it is added to the property `matches.approved` in
+    obj.extra_data.
```
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/matching.py]: I think that the name should be changed, according to this. I'd expect a step called 'is_...' not to change anything.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 4]
```diff
@@ -98,12 +98,87 @@ def article_exists(obj, eng):
...
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the property fuzzy_match_approved_id
+        in obj.extra_data contains an integer id, ``False`` otherwise.
+    """
+    approved_match_id = obj.extra_data.get('fuzzy_match_approved_id')
+    obj.extra_data.setdefault('matches', {})['approved'] = approved_match_id
+    return bool(isinstance(approved_match_id, (int, long)))
```
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/matching.py]: the bool is not needed here, isinstance always returns bool.

Comment (david-caro) [reply inspirehep/modules/workflows/tasks/matching.py]: The fact that you are checking the type of the value looks strange. Should we care about that? If it's a string or anything? I think we should only care if there's any, not what it is.

File: inspirehep/modules/workflows/workflows/article.py [hunk 5]
```diff
@@ -392,11 +394,32 @@
...
-        mark('is-update', False),
+        exact_match,
+        [
+            mark('is-update', True),
+            mark('exact-matched', True),
+        ],
+        IF_ELSE(
+            fuzzy_match,
+            [
+
```
Comment (david-caro) [inline inspirehep/modules/workflows/workflows/article.py]: extra newline

Comment (david-caro) [reply inspirehep/modules/workflows/workflows/article.py]: If you added it to make it more readable, I suggest using a new array var instead (the equivalent of moving to a new function).

File: inspirehep/modules/workflows/tasks/matching.py (hunk 3 above)
Comment (david-caro) [reply inspirehep/modules/workflows/tasks/matching.py]: Maybe we can split it in `handle_fuzzy_match_decision` that does the changes in the workflow, and `is_there_an_approval`.

The second one might not care about if the approval is fuzzy match or not, but just that there's a result in the `matches.approved`.

Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: Do you mean in the future, we may want to halt even for exact matches and reuse this part? Sounds reasonable

Comment (david-caro) [reply inspirehep/modules/workflows/tasks/matching.py]: Yep, but I was focusing more on those two actions being different things.

File: inspirehep/modules/workflows/workflows/article.py (hunk 5 above)
Comment (ammirate) [reply inspirehep/modules/workflows/workflows/article.py]: just extra

File: inspirehep/modules/workflows/tasks/matching.py (hunk 4 above)
Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: Depends on how we define the API. If we specify the front-end has to pass an integer, then we can check if it is compliant to the spec.

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 6]
```diff
@@ -23,21 +23,25 @@
 from __future__ import absolute_import, division, print_function
 
 from mock import patch
+from inspire_utils.record import get_value
 
 from inspire_schemas.api import load_schema, validate
 from inspirehep.modules.workflows.tasks.matching import (
-    article_exists,
     has_fully_harvested_category,
     physics_data_an_is_primary_category,
```
Comment (iulianav) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: These are not in order.

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 7]
```diff
@@ -279,3 +283,79 @@ def test_pending_in_holding_pen_returns_false_if_nothing_matched(mock_match):
...
+    assert 'matches' in obj.extra_data
+
+    expected = []
+    result = get_value(obj.extra_data, 'matches.fuzzy')
+
+    assert expected == result
+
+
+@patch('inspirehep.modules.workflows.tasks.matching.match')
+def test_is_fuzzy_match_approved_returns_returns_true_if_there_is_a_match_approved(mock_match):
```
Comment (iulianav) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: 2 'returns' in the test name.

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 8]
```diff
@@ -279,3 +283,79 @@ def test_pending_in_holding_pen_returns_false_if_nothing_matched(mock_match):
...
+    assert 'matches' in obj.extra_data
+
+    expected = 4328
+    result = get_value(obj.extra_data, 'matches.approved')
+
+    assert expected == result
+
+
+@patch('inspirehep.modules.workflows.tasks.matching.match')
+def test_is_fuzzy_match_approved_returns_returns_False_if_there_is_not_a_match_approved(mock_match):
```
Comment (iulianav) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: Again 2 'returns'.

File: tests/unit/workflows/test_workflows_actions.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from invenio_workflows import ObjectStatus
+
+from inspirehep.modules.workflows.actions import MatchApproval, MergeApproval
+from mocks import MockObj
```
Comment (iulianav) [inline tests/unit/workflows/test_workflows_actions.py]: I think this should be placed right after the '__future__' import.

File: tests/integration/workflows/test_workflows_manual_merge.py [hunk 10]
```diff
@@ -25,7 +25,7 @@
 import pytest
 
 from invenio_workflows import ObjectStatus, workflow_object_class
-from invenio_records.api import RecordMetadata
+from invenio_records.models import RecordMetadata
```
Comment (iulianav) [inline tests/integration/workflows/test_workflows_manual_merge.py]: These are not in alphabetical order.

File: tests/acceptance/conftest.py [hunk 11]
```diff
@@ -116,3 +117,11 @@ def cleanup_workflows_tables(app):
                 obj.delete()
 
         db.session.commit()
+
+
+@pytest.fixture(autouse=True, scope='function')
+def cleanup_literatures(app):
+    recs = RecordMetadata.query.filter(RecordMetadata.json['_collections'].op('?')('Literature')).all()
```
Comment (iulianav) [inline tests/acceptance/conftest.py]: Can't you maybe do a straight 'delete' here with a filter or smth? I am not sure.

File: inspirehep/modules/workflows/actions/__init__.py [hunk 12]
```diff
@@ -27,3 +27,4 @@
 from .hep_approval import HEPApproval        # noqa: F401
 from .merge_approval import MergeApproval    # noqa: F401
 from .author_approval import AuthorApproval  # noqa: F401
+from .match_approval import MatchApproval  # noqa: F401
```
[the remark this answers is not in the document]
Comment (ammirate) [reply inspirehep/modules/workflows/actions/__init__.py]: :P

File: inspirehep/modules/workflows/tasks/upload.py [hunk 13]
```diff
@@ -45,7 +46,7 @@ def _get_updated_record(obj):
             )
         else:
             pid_type = get_pid_type_from_schema(obj.data['$schema'])
-            updated_record_id = obj.extra_data['record_matches'][0]
+            updated_record_id = get_value(obj.extra_data, 'matches.approved')
```
Comment (drjova) [inline inspirehep/modules/workflows/tasks/upload.py]: we've to double check if the same behaviour for ``exact_match`` and ``fuzzy_match``

File: inspirehep/modules/workflows/tasks/matching.py [hunk 14]
```diff
@@ -98,12 +98,77 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
     record_ids = [el['_source']['control_number'] for el in matches]
-    if record_ids:
-        obj.extra_data['record_matches'] = record_ids
-        return True
+    obj.extra_data.setdefault('matches', {})['exact'] = record_ids
```
Comment (drjova) [inline inspirehep/modules/workflows/tasks/matching.py]: has to be the first record of the list in ``matched.approved``

Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: Great!

File: inspirehep/modules/workflows/mappings/holdingpen/hep.json [hunk 15]
```diff
@@ -12,7 +12,7 @@
                         "is-update": {
                             "type": "boolean"
                         },
-                        "record_matches": {
+                        "matches": {
```
Comment (jacquerie) [inline inspirehep/modules/workflows/mappings/holdingpen/hep.json]: OPS: the content of this key should be migrated as well.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 16]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+                            },
+                        ],
+                        'type': 'fuzzy',
+                    }
+                ]
+            }
+        ],
+        'doc_type': 'hep',
+        'index': 'records-hep',
+    }
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: This should be part of the configuration, so that it can be overridden in the OPS repo (if need arises).

File: inspirehep/modules/workflows/tasks/matching.py [hunk 17]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
     record_ids = [el['_source']['control_number'] for el in matches]
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: Since https://github.com/inspirehep/inspire-matcher/commit/3d9cec0df0944acd7733949e776f1e95248a13be one can avoid this and just pass the `source` parameter to the `match` call to do filtering already on ES (which is much faster, as it reduces network communication).

File: inspirehep/modules/workflows/tasks/matching.py [hunk 18]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+                    }
+                ]
+            }
+        ],
+        'doc_type': 'hep',
+        'index': 'records-hep',
+    }
+
+    matches = dedupe_list(match(obj.data, fuzzy_match_configuration))
+    record_ids = [el['_source']['control_number'] for el in matches]
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: Same as https://github.com/inspirehep/inspire-next/pull/3129/files#r171280155.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 19]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
     """
     matches = dedupe_list(match(obj.data))
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: I was sloppy here, and relied on the default configuration of the matcher, but it should be passed as a parameter (fetched from the configuration, as observed in https://github.com/inspirehep/inspire-next/pull/3129/files#r171278799).

File: inspirehep/modules/workflows/tasks/matching.py [hunk 20]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+
+
+def set_fuzzy_match_approved_in_extradata(obj, eng):
+    """Set the human approved match in `matches.approved` in extra_data."""
+    approved_match = obj.extra_data.get('fuzzy_match_approved_id')
+    obj.extra_data.setdefault('matches', {})['approved'] = approved_match
+
+
+def set_exact_match_as_approved_in_extradata(obj, eng):
+    """Set the human approved match in `matches.approved` in extra_data."""
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: Rogue Ctrl-C Ctr-V ?

File: inspirehep/modules/workflows/tasks/matching.py [hunk 21]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+    best_match = get_value(obj.extra_data, 'matches.exact')[0]
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: You can just `get_value(obj.extra_data, 'matches.exact[0]')`.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 22]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+    return bool(record_ids)
+
+
+@with_debug_logging
+def is_fuzzy_match_approved(obj, eng):
+    """Check if a fuzzy match has been approved by a human."""
+    return obj.extra_data.get('fuzzy_match_approved_id')
+
+
+def set_fuzzy_match_approved_in_extradata(obj, eng):
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: Missing `with_debug_logging` decorator.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 23]
```diff
@@ -98,12 +98,83 @@ def article_exists(obj, eng):
...
+    """Set the human approved match in `matches.approved` in extra_data."""
+    approved_match = obj.extra_data.get('fuzzy_match_approved_id')
+    obj.extra_data.setdefault('matches', {})['approved'] = approved_match
+
+
+def set_exact_match_as_approved_in_extradata(obj, eng):
```
Comment (jacquerie) [inline inspirehep/modules/workflows/tasks/matching.py]: Missing `with_debug_logging` decorator.

File: tests/acceptance/conftest.py (hunk 11 above)
Comment (ammirate) [reply tests/acceptance/conftest.py]: mmh nope, the result of the query relies on the _collection field, which is in the json column of the table

File: inspirehep/modules/workflows/tasks/matching.py (hunk 16 above)
Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: Moving it in `inspirehep/config.py`.

File: inspirehep/modules/workflows/tasks/matching.py (hunk 17 above)
Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: The filtering works as expected, but it still returns a minimal object, from which we have to extract only the `control_number`, so this statement is still required.

This is the object returned by `match`:
```python
{
  u'_id': u'c672d49e-f633-45dd-89d3-78d653098350',
  u'_index': u'records-hep',
  u'_score': 1.292094,
  u'_source': {u'control_number': 1234},
  u'_type': u'hep'
}
```

File: inspirehep/modules/workflows/tasks/upload.py (hunk 13 above)
Comment (david-caro) [reply inspirehep/modules/workflows/tasks/upload.py]: Break if the approved is None or not there (key not there).

File: inspirehep/modules/workflows/workflows/article.py [hunk 24]
```diff
@@ -395,11 +399,33 @@
 ]
 
 
-MARK_IF_UPDATE = [
+HALT_FOR_MATCH_APPROVAL_OR_MARK_IF_UPDATE = [
```
Comment (david-caro) [inline inspirehep/modules/workflows/workflows/article.py]: CHECK_IS_UPDATE

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 25]
```diff
@@ -76,11 +78,12 @@
     return_value=[],
 )
 def test_harvesting_arxiv_workflow_manual_rejected(
-    mocked_arxiv_download,
     mocked_refextract_extract_refs,
     mocked_api_request_magpie,
-    mocked_api_request_beard,
     mocked_package_download,
```
Comment (david-caro) [inline tests/integration/workflows/test_arxiv_workflow.py]: s/mocked_package_download/mocked_beard_api or whatever

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 26]
```diff
@@ -136,11 +140,10 @@ def test_harvesting_arxiv_workflow_manual_rejected(
     return_value=[],
 )
 def test_harvesting_arxiv_workflow_core_record_auto_accepted(
-    mocked_download,
-    mocked_is_pdf,
```
Comment (david-caro) [inline tests/integration/workflows/test_arxiv_workflow.py]: Looks like you removed one too many, strange it did not fail though (checking locally).

Comment (david-caro) [reply tests/integration/workflows/test_arxiv_workflow.py]: Tested locally, yes, it's overriding the workflow_app. My guess is that one of the other fixtures (probably mocked_external_services) is already requiring the workflow app, and it's doing so inside an app context, so the magick mock `workflow_app.app_context()` is not really doing anything, and it does not fail just because it's lucky.

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 27]
```diff
@@ -261,10 +263,11 @@ def test_harvesting_arxiv_workflow_manual_accepted(
     side_effect=fake_magpie_api_request,
 )
 def test_match_in_holdingpen_stops_pending_wf(
-    mocked_download_arxiv,
-    mocked_api_request_beard,
     mocked_api_request_magpie,
+    mocked_api_request_beard,
     mocked_package_download,
+    mocked_is_pdf_link,
+    mocked_download_arxiv,
```
Comment (david-caro) [inline tests/integration/workflows/test_arxiv_workflow.py]: This one was using one less param as needed... it's the reverse case than the other one :), now it's ok.

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 28]
```diff
@@ -498,3 +503,109 @@ def test_article_workflow_continues_when_record_is_valid(workflow_app):
...
+    mocked_api_request_beard,
+    mocked_api_request_magpie,
+    workflow_app,
+    mocked_external_services,
+    record_from_db,
+):
+    """Test update article fuzzy matched.
+
+    In this test the `WorkflowObject.continue_workflow` is mocked because in
+    the test suite celery is run in eager mode. This prevent celery to switch
```
Comment (david-caro) [inline tests/integration/workflows/test_arxiv_workflow.py]: s/prevent/prevents/

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 29]
```diff
@@ -498,3 +503,109 @@ def test_article_workflow_continues_when_record_is_valid(workflow_app):
...
+    assert obj.extra_data['already-in-holding-pen'] is False
+    assert obj.extra_data['holdingpen_matches'] == []
+    assert obj.extra_data['matches']['fuzzy'] == [rec_id]
+
+    WorkflowObject.continue_workflow = continue_wf_patched_context(workflow_app)
+    do_resolve_matching(workflow_app, obj.id, rec_id)
+
+    obj = workflow_object_class.get(obj_id)
+    assert obj.extra_data['matches']['approved'] == rec_id
+    assert obj.extra_data['is-update']
```
Comment (david-caro) [inline tests/integration/workflows/test_arxiv_workflow.py]: Should this be approved/completed? It's not really clear to me what we are testing here though :/, we tend to mix integration with end-to-end with acceptance with ui...

File: tests/integration/workflows/test_workflows_manual_merge.py (hunk 10 above)
Comment (david-caro) [inline tests/integration/workflows/test_workflows_manual_merge.py]: why this change?

File: tests/integration/workflows/test_workflows_manual_merge.py [hunk 30]
```diff
@@ -73,7 +71,12 @@ def test_manual_merge_existing_records(workflow_app):
         current_user_id=1,
     )
 
-    do_resolve_manual_merge_wf(workflow_app, obj_id)
+    # assuming MergeApproval.resolve() is called by the api
```
Comment (david-caro) [inline tests/integration/workflows/test_workflows_manual_merge.py]: and this change? you adapted the function in https://github.com/inspirehep/inspire-next/pull/3129/files#diff-45d44c66955f78a020f91bbab856b800R65

File: tests/acceptance/conftest.py (hunk 11 above)
Comment (david-caro) [reply tests/acceptance/conftest.py]:
```python
RecordMetadata.query.filter(RecordMetadata.json['_collections'].op('?'), 'Literature')).delete(synchronize_session=False)
```

check http://docs.sqlalchemy.org/en/latest/orm/query.html?highlight=delete#sqlalchemy.orm.query.Query.delete for details.

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 31]
```diff
@@ -23,40 +23,45 @@
 from __future__ import absolute_import, division, print_function
 
 from mock import patch
+from inspire_utils.record import get_value
 
 from inspire_schemas.api import load_schema, validate
 from inspirehep.modules.workflows.tasks.matching import (
-    article_exists,
+    _pending_in_holding_pen,
```
Comment (david-caro) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: This should be renamed to not `_`, or not taken into account in the tests imo.

Comment (david-caro) [reply tests/unit/workflows/test_workflows_tasks_matching.py]: I was there already though :/

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 32]
```diff
@@ -23,40 +23,45 @@
...
 )
 
 from mocks import MockEng, MockObj
 
 
 @patch('inspirehep.modules.workflows.tasks.matching.match')
-def test_article_exists_returns_true_if_something_matched(mock_match):
-    mock_match.return_value = iter([{'_source': {'control_number': 4328}}])
+def test_exact_match_returns_true_if_something_matched(mock_match):
+    mock_match.return_value = iter([{'_source': {'control_number': 4238}}])
```
Comment (david-caro) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: Is this control number relevant?

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 33]
```diff
@@ -65,11 +70,11 @@ def test_article_exists_returns_false_if_nothing_matched(mock_match):
     obj = MockObj(data, extra_data)
     eng = MockEng()
 
-    assert not article_exists(obj, eng)
-    assert 'record_matches' in obj.extra_data
+    assert not exact_match(obj, eng)
+    assert 'matches' in obj.extra_data
```
Comment (david-caro) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: It feels strange to me that you are checking that `'matches'` is in `extra_data` but not `'exact'`.

File: tests/integration/workflows/test_arxiv_workflow.py (hunk 29 above)
Comment (ammirate) [reply tests/integration/workflows/test_arxiv_workflow.py]: This is testing a new branch of the workflow: when the fuzzy matching branch is walked through, some things happen in the workflow (marks, halting).

File: tests/integration/workflows/test_workflows_manual_merge.py (hunk 30 above)
Comment (ammirate) [reply tests/integration/workflows/test_workflows_manual_merge.py]: leftover of a previous refactor T.T

File: tests/unit/workflows/test_workflows_tasks_matching.py (hunk 33 above)
Comment (ammirate) [reply tests/unit/workflows/test_workflows_tasks_matching.py]: there is this assertion after that is checking it, isn't it ok?
`result = get_value(obj.extra_data, 'matches.exact')`

File: inspirehep/modules/workflows/tasks/matching.py (hunk 16 above)
Comment (david-caro) [reply inspirehep/modules/workflows/tasks/matching.py]: Something happened here... wasn't this config moved?

File: tests/integration/workflows/test_workflows_manual_merge.py (hunk 10 above)
Comment (ammirate) [reply tests/integration/workflows/test_workflows_manual_merge.py]: Wrong refactor :+1:

**Comment by david-caro** [comment]
ci merge please

**Comment by david-caro** [comment]
ci test please

**Comment by ammirate** [comment]
ci test please

**Comment by ammirate** [comment]
ci merge please
